File: artwalk_backend/src/main/java/com/ssafy/a401/artwalk_backend/domain/route/repository/RouteRepository.java
package com.ssafy.a401.artwalk_backend.domain.route.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import com.ssafy.a401.artwalk_backend.domain.route.model.Route;

public interface RouteRepository extends JpaRepository<Route, Integer> {
	long countByRouteId(int routeId);

	long countByUserId(String userId);

	List<Route> findByUserIdOrderByRouteIdDesc(String userId);

	List<Route> findByUserIdContainingOrderByRouteIdDesc(String userId);

	List<Route> findByMakerContainingOrderByRouteIdDesc(String maker);

	List<Route> findByTitleContainingOrderByRouteIdDesc(String title);
}

File: Artwalk_Android/domain/src/main/java/com/a401/domain/repository/RouteRepository.kt
package com.a401.domain.repository

import com.a401.domain.model.Marker
import com.a401.domain.model.RouteForDraw
import com.a401.domain.model.RouteForList

interface RouteRepository {
    suspend fun getRouteForWalking(from: Marker, to: Marker): Result<RouteForDraw>

    suspend fun getRoutForList(): List<RouteForList>
}